### ISSUE TITLE
Macro importing from a template does not work with template inheritance

### ISSUE THREAD

**Issue by hason** [twigphp/Twig#1212]
To reproduce the bug:

``` jinja
{# index.twig #}
{% extends "parent.twig" %}
{% block content %}
    {{ hello("Jack") }}
{% endblock %}

{# parent.twig #}
{% from "macros.twig" import hello %}
{% block content %}{% endblock %}

{# macros.twig #}
{% macro hello(name) %}
    Hello {{ name }}
{% endmacro %}
```

Parsing fails:

```
Twig_Error_Syntax: The function "hello" does not exist in "index.twig"
```

**Comment by fabpot** [comment]
I think this one should be closed as #1217 should not work and will be fixed in 2.0.

**Comment by fabpot** [comment]
It's indeed expected as macros imports are local to a template, not something we will change, so closing.
